fem/nonlininteg.cpp

```diff
@@ -385,4 +439,230 @@ void HyperelasticNLFIntegrator::AssembleElementGrad(const FiniteElement &el,
    }
 }
 
+double IncompressibleNeoHookeanIntegrator::GetElementEnergy(
+   const Array<const FiniteElement *>&el,
+   ElementTransformation &Tr,
+   const Array<const Vector *>&elfun)
+{
+   if (el.Size() != 2)
+   {
+      mfem_error("IncompressibleNeoHookeanIntegrator::GetElementEnergy"
+                 " has incorrect block finite element space size!");
+   }
+
+   int dof_u = el[0]->GetDof();
+   int dim = el[0]->GetDim();
+
+   DSh_u.SetSize(dof_u, dim);
+   J0i.SetSize(dim);
+   J1.SetSize(dim);
+   J.SetSize(dim);
```

Comment:
Do we want to support surface meshes embedded in 3D or is that a feature for another day?  If so spaceDim!=dim

Reply:
Manifold problems don't really make sense for these hyperelastic models as the 2D case only really makes sense if you consider it to be in plane strain. For the manifold case, you would want a full shell or plate model. I'll add a check to make sure it's not trying to be used in that way.